indico/web/flask/errors.py

```diff
@@ -42,6 +42,19 @@ def handle_forbidden(exc):
     return render_error(exc, _('Access Denied'), get_error_description(exc), exc.code)
 
 
+@errors_bp.app_errorhandler(UnprocessableEntity)
+def handle_unprocessableentity(exc):
+    data = getattr(exc, 'data', None)
+    if data and 'messages' in data and (request.is_xhr or request.is_json):
```

Comment:
it's only deprecated in the flask core because many recent ajax libraries don't add it by default (it was mainly a prototype/jquery thing). if it shows a deprecation warning, we can simply override it in our IndicoRequest class to just do the check (or even use a different header if we want to)